File: src/greedy/B_1715.java
package greedy;

import java.util.*;
import java.io.*;

public class B_1715 {

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int N = Integer.parseInt(br.readLine());
		
		PriorityQueue<Integer> pq = new PriorityQueue<>();
		for(int i=0;i<N;i++)
			pq.add(Integer.parseInt(br.readLine()));
		
		int answer = 0;
		while(pq.size() > 1) {
			int first = pq.poll();
			int second = pq.poll();
			
			pq.add(first + second);
			answer += (first + second);
		}
		
		System.out.println(answer);
	}

}

File: src/kotlin/greedy/B_1715.kt
import java.util.*

fun main() = with(System.`in`.bufferedReader()) {
    val n = readLine().toInt()
    val pq = PriorityQueue<Int>()
    repeat(n) {
        pq.offer(readLine().toInt())
    }

    var answer = 0
    while (pq.size > 1) {
        val first = pq.poll()
        val second = pq.poll()
        answer += first + second
        pq.offer(first + second)
    }
    print(answer)
}
